src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPattern.java

```diff
@@ -1,102 +1,117 @@
 package edu.agh.wfiis.designpatterns.patternsUsageExample;
 
-class DesignPattern {
+import org.springframework.stereotype.Component;
 
-    private String id;
-
-    private String name;
+public class DesignPattern {
+    private Details patternDetails;
 
     private String patternCode;
 
-    private String description;
-
-    private String example;
-
-    private boolean creational;
-
-    private boolean structural;
-
-    private boolean behavioral;
-
-    public String getId() {
-        return id;
+    public static DesignPattern create(CreatingDesignPattern creator) {
```

Comment:
By exposing only one public constructor which takes param of type with private constructor that makes this type uncreatable from outside code, and finally makes DesignPattern (beside 3 predefined combinations) type uncreatable.